### ISSUE TITLE
Windows binary release on Chocolatey?

### ISSUE THREAD

**Issue by Jalalx** [k0sproject/k0s#571]
[Chocolatey](https://chocolatey.org/) is a Windows-based package manager. It would be awesome if we could have k0s binary available in the Chocolatey!

**Comment by jnummelin** [comment]
@Jalalx I do not have any experience on Chocolatey, could you summarize a bit what it would take to get k0s there?

**Comment by Jalalx** [comment]
Chocolatey is a package manager that makes installing any Windows-based application easy. It is like `apt` in Ubuntu or Homebrew in MacOS.

Here are documentations about creating a Chocolatey package: https://docs.chocolatey.org/en-us/create/
I am not an expert on creating Chocolatey packages, but basically, the requirements are:

1. Creating a [Chocolatey account](https://chocolatey.org/account/Register) and obtaining an API Key [on the page bottom](https://chocolatey.org/account).
2. Installing Chocolatey itself on your Windows client
3. Running `choco new k0s-unofficial ...` command to build the package template
4. Filling the configuration files & copying the `k0s.exe` binary file to the `/tools` directory
5. `choco apikey --key <your API Key goes here> --source https://push.chocolatey.org/` But this is only once per machine and needs Administrator privilates.
5. `choco pack --version 0.9.0` Creates the Chocoaltey Package named `k0s-unofficial.0.9.0.nupkg`
6. `choco push k0s-unofficial.0.9.0.nupkg --source https://push.chocolatey.org`

Here is a package I've already pushed: https://chocolatey.org/packages/k0s-unofficial/0.9.0

**Comment by Jalalx** [comment]
And here is the `k0s-unofficial.nuspec` file content:
<details>
  <summary>Click to expand file content</summary>

```xml
<?xml version="1.0" encoding="utf-8"?>
<!-- Read this before creating packages: https://chocolatey.org/docs/create-packages -->
<!-- It is especially important to read the above link to understand additional requirements when publishing packages to the community feed aka dot org (https://chocolatey.org/packages). -->

<!-- Test your packages in a test environment: https://github.com/chocolatey/chocolatey-test-environment -->

<!--
This is a nuspec. It mostly adheres to https://docs.nuget.org/create/Nuspec-Reference. Chocolatey uses a special version of NuGet.Core that allows us to do more than was initially possible. As such there are certain things to be aware of:

* the package xmlns schema url may cause issues with nuget.exe
* Any of the following elements can ONLY be used by choco tools - projectSourceUrl, docsUrl, mailingListUrl, bugTrackerUrl, packageSourceUrl, provides, conflicts, replaces 
* nuget.exe can still install packages with those elements but they are ignored. Any authoring tools or commands will error on those elements 
-->

<!-- You can embed software files directly into packages, as long as you are not bound by distribution rights. -->
<!-- * If you are an organization making private packages, you probably have no issues here -->
<!-- * If you are releasing to the community feed, you need to consider distribution rights. -->
<!-- Do not remove this test for UTF-8: if “Ω” doesn’t appear as greek uppercase omega letter enclosed in quotation marks, you should use an editor that supports UTF-8, not this one. -->
<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
  <metadata>
    <!-- == PACKAGE SPECIFIC SECTION == -->
    <!-- This section is about this package, although id and version have ties back to the software -->
    <!-- id is lowercase and if you want a good separator for words, use '-', not '.'. Dots are only acceptable as suffixes for certain types of packages, e.g. .install, .portable, .extension, .template -->
    <!-- If the software is cross-platform, attempt to use the same id as the debian/rpm package(s) if possible. -->
    <id>k0s-unofficial</id>
    <!-- version should MATCH as closely as possible with the underlying software -->
    <!-- Is the version a prerelease of a version? https://docs.nuget.org/create/versioning#creating-prerelease-packages -->
    <!-- Note that unstable versions like 0.0.1 can be considered a released version, but it's possible that one can release a 0.0.1-beta before you release a 0.0.1 version. If the version number is final, that is considered a released version and not a prerelease. -->
    <version>0.0.0</version>
    <packageSourceUrl>https://github.com/k0sproject/k0s</packageSourceUrl>
    <!-- owners is a poor name for maintainers of the package. It sticks around by this name for compatibility reasons. It basically means you. -->
    <!--<owners>__REPLACE_YOUR_NAME__</owners>-->
    <!-- ============================== -->

    <!-- == SOFTWARE SPECIFIC SECTION == -->
    <!-- This section is about the software itself -->
    <title>k0s-unofficial (Install)</title>
    <authors>k0s Team</authors>
    <!-- projectUrl is required for the community feed -->
    <projectUrl>https://k0sproject.io</projectUrl>
    <iconUrl>https://k0sproject.io/images/k0s-logo.svg</iconUrl>
    <!-- <copyright>Year Software Vendor</copyright> -->
    <!-- If there is a license Url available, it is required for the community feed -->
    <!-- <licenseUrl>Software License Location __REMOVE_OR_FILL_OUT__</licenseUrl>
    <requireLicenseAcceptance>true</requireLicenseAcceptance>-->
    <projectSourceUrl>https://github.com/k0sproject/k0s</projectSourceUrl>
    <!--<docsUrl>At what url are the software docs located?</docsUrl>-->
    <!--<mailingListUrl></mailingListUrl>-->
    <bugTrackerUrl>https://github.com/k0sproject/k0s/issues</bugTrackerUrl>
    <tags>k0s-unofficial k0s kubernetes</tags>
    <summary>Deploy and run Kubernetes workloads at any scale on any infrastructure.
All batteries included. 100% open source &amp; free.</summary>
    <description>k0s is the simple, solid &amp; certified Kubernetes distribution that works on any infrastructure: public &amp; private clouds, on-premises, edge &amp; hybrid. It's 100% open source &amp; free. Get started today!</description>
    <!-- <releaseNotes>__REPLACE_OR_REMOVE__MarkDown_Okay</releaseNotes> -->
    <!-- =============================== -->      

    <!-- Specifying dependencies and version ranges? https://docs.nuget.org/create/versioning#specifying-version-ranges-in-.nuspec-files -->
    <!--<dependencies>
      <dependency id="" version="__MINIMUM_VERSION__" />
      <dependency id="" version="[__EXACT_VERSION__]" />
      <dependency id="" version="[_MIN_VERSION_INCLUSIVE, MAX_VERSION_INCLUSIVE]" />
      <dependency id="" version="[_MIN_VERSION_INCLUSIVE, MAX_VERSION_EXCLUSIVE)" />
      <dependency id="" />
      <dependency id="chocolatey-core.extension" version="1.1.0" />
    </dependencies>-->
    <!-- chocolatey-core.extension - https://chocolatey.org/packages/chocolatey-core.extension
         - You want to use Get-UninstallRegistryKey on less than 0.9.10 (in chocolateyUninstall.ps1)
         - You want to use Get-PackageParameters and on less than 0.11.0
         - You want to take advantage of other functions in the core community maintainer's team extension package
    -->

    <!--<provides>NOT YET IMPLEMENTED</provides>-->
    <!--<conflicts>NOT YET IMPLEMENTED</conflicts>-->
    <!--<replaces>NOT YET IMPLEMENTED</replaces>-->
  </metadata>
  <files>
    <!-- this section controls what actually gets packaged into the Chocolatey package -->
    <file src="tools\**" target="tools" />
    <!--Building from Linux? You may need this instead: <file src="tools/**" target="tools" />-->
  </files>
</package>
```
</details>

**Comment by jnummelin** [comment]
The issue is marked as stale since no activity has been recorded in 30 days
